iothub_client/src/iothub_client_properties.c

```diff
@@ -69,137 +71,159 @@ static void AdvanceCountersAfterWrite(size_t currentOutputBytes, char** currentW
     }
 }
 
+// To correctly check snprintf for success, we need to check both it's not negative AND 
+// that the (number of bytes written ) != (sizeof buffer).
+static int PropertySnprintf(char* buffer, size_t count, const char* format, ...)
+{
+    va_list arg_list;
+    va_start(arg_list, format);
+
+    int currentOutputBytes = vsnprintf(buffer, count, format, arg_list);
+    if ((currentOutputBytes < 0) || (currentOutputBytes == (int)count))
+    {
+        return -1;
+    }
+
+    return currentOutputBytes;
+}
+
+
 // WriteOpeningBrace adds opening brace for JSON to be generated, either a simple "{" or more complex format if componentName is specified.
-static size_t WriteOpeningBrace(const char* componentName, char* currentWrite, size_t remainingBytes)
+static bool WriteOpeningBrace(const char* componentName, char** currentWrite, size_t* requiredBytes, size_t* remainingBytes)
 {
-    size_t currentOutputBytes;
+    int currentOutputBytes;
 
     if (componentName != NULL)
     {
-        if ((currentOutputBytes = snprintf(currentWrite, remainingBytes, PROPERTY_FORMAT_COMPONENT_START, componentName)) < 0)
+        if ((currentOutputBytes = PropertySnprintf(*currentWrite, *remainingBytes, PROPERTY_FORMAT_COMPONENT_START, componentName)) < 0)
```

Comment:
Nit: Curious why checking for general negative int instead of specific -1 expected return value on failure ?

Reply:
It'd work with -1 as implemented.  But I'm trying to keep this `snprintf` wrapper's caller as close to the original `snprintf` as possible, and on the C API you check negative and not the exact value.